Firestore/Source/API/FIRFirestore.mm

```diff
@@ -16,8 +16,13 @@
 
 #import "FIRFirestore.h"
 
+#import <FirebaseAuthInterop/FIRAuthInterop.h>
```

Comment:
This file is now getting huge. Is there any way we can split out these other things into separate parts?

Ideally you should be able to reason about FIRFirestore without thinking about interop stuff and be able to reason about the interop stuff separately too.

Reply:
Chatted offline, but adding it here for the record: moving the interop code to a new class called `FSTFirestoreComponent` that will handle registration and instance creation/retrieval.